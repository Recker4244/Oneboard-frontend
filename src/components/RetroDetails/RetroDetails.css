.retro-form-heading {
  display: flex;
  background-color: #f7fbff;
  height: 9%;
  margin-bottom: 10px;
  padding: 20px;
  align-items: center;
  justify-content: space-between;
}
.retro-form-heading div {
  display: flex;
  flex-direction: row;
  font-size: 21px;
  /* font-weight: 500; */
  /* color: #030c82; */
  padding-left: 0px;
  width: 176px;
  height: 38px;
  /* overflow: hidden; */
}

.retro-title-left-icon div{
  color: #030c82;
}

.retro-title-left-icon {
  display: flex;
  align-items: center;
  color: grey;
}

.retro-title-left-icon button {
  margin-left: 10px; /* or any desired spacing */
}


.edit-button-retro {
  position: relative;
  bottom: 14px;
  left: 10px;
  color: grey;
  font-size: 31px;
  width: 62px;
  cursor: pointer;
}
.cancel-button-retro {
  position: relative;
  right: 19px;
  font-size: 25px;
  font-weight: bolder;
}

.retro-form-body-details {
  display: flex;
  flex-direction: row;
  padding: 50px;
  height: 90%;
  width: 100%;
}

.retro-form-body-details .image {
  width: 60%;
  height: 90%;
  border-right: 1px solid lightgrey;
  padding-right: 30px;
  margin-right: 30px;
}

.image img {
  height: 100%;
  width: 100%;
}

.action-items-details {
  margin-left: 42px;
}

.actions {
  list-style: circle;
  list-style-type: disc;
  margin-top: 10px;
}

.retro-details-left {
  object-fit: fill;
}

.retro-details-right h1 {
  font-size: 25px;
  padding-left: 30px;
}

.retro-form-content {
  display: flex;
  flex-direction: column;
  width: 784.8px;
  height: 553.8px;
}

@media screen and (max-width: 480px) {
  .retro-form-heading {
    background-color: #f7fbff;
    height: 15%;
    margin-bottom: 10px;
    padding: 29px;
    padding-left: 30px;
  }
  .cancel-button-retro {
    position: relative;
    font-size: 25px;
    font-weight: bolder;
    left: 89%;
    bottom: 39px;
  }
  .retro-form-body-details {
    display: flex;
    flex-direction: column;
    padding: 50px;
    height: 90%;
    width: 100%;
  }
  .retro-form-body-details .image {
    width: 60%;
    height: 90%;
    border-right: 0px;
    padding-right: 30px;
    margin-right: 30px;
  }
  .retro-form-body-details .image {
    width: 114%;
    height: 90%;
  }
  .retro-form-body-details .image {
    border-right: 0px;
    padding-right: 20px;
    margin-right: 0px;
    width: 114%;
    height: 90%;
  }
}
@media screen and (max-width: 1120px) {
  .retro-details-right h1 {
    font-size: 20px;
    font-weight: 700px;
    padding-left: 30px;
  }
  .image img {
    height: 95%;
    width: 100%;
  }
  .retro-form-heading {
    background-color: #f7fbff;
    height: 11%;
    margin-bottom: 10px;
    padding: 29px;
    padding-left: 30px;
  }
  .actions {
    list-style: circle;
    list-style-type: disc;
    margin-top: 10px;
    font-weight: 300;
    font-size: 20px;
  }
}

@media screen and (max-width: 962px) {
  .retro-details-right h1 {
    font-size: 19px;
    font-weight: 700px;
    padding-left: 30px;
  }
  .image img {
    height: 90%;
    width: 100%;
  }
  .actions {
    list-style: circle;
    list-style-type: disc;
    margin-top: 10px;
    font-size: 15px;
  }
  retro-form-heading {
    background-color: #f7fbff;
    height: 15%;
    margin-bottom: 10px;
    padding: 29px;
    padding-left: 30px;
  }
}

@media screen and (min-width: 735) and (max-width: 948px) {
  .retro-details-right h1 {
    font-size: 12px;
    font-weight: 700px;
    padding-left: 30px;
  }
  .image img {
    height: 90%;
    width: 100%;
  }
  .actions {
    list-style: circle;
    list-style-type: disc;
    margin-top: 10px;
    font-size: 15px;
  }
}



    