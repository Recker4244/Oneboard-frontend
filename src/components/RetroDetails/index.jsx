/* eslint-disable max-len */
/* eslint-disable jsx-a11y/control-has-associated-label */
import React from 'react';
import './RetroDetails.css';
import PropTypes from 'prop-types';
import CloseRoundedIcon from '@mui/icons-material/CloseRounded';
import EditIcon from '@mui/icons-material/Edit';
import RetroUpdate from '../RetroUpdate';

function RetroDetails(props) {
  const {
    retroBoard, click, setClick, role
  } = props;
  const [edit, setEdit] = React.useState(false);
  const [imageSrc, setImageSrc] = React.useState(`${retroBoard.image_path}`);
  const [imageNotFound, setImageNotFound] = React.useState(false);

  const handleImageError = () => {
    if (!imageNotFound) {
      setImageNotFound(true);
      setImageSrc('https://st3.depositphotos.com/23594922/31822/v/600/depositphotos_318221368-stock-illustration-missing-picture-page-for-website.jpg');
    }
  };
  // console.log(retroBoard);
  // console.log(role);
  // console.log(click);
  const editRetroHandler = () => {
    setEdit(!edit);
  };

  const handleClick = () => {
    setClick(!click);
  };
  return (
    <div className="retro-form z-50">
      <div className="retro-form-content">
        <div className="retro-form-heading">
          <div className="retro-title-left-icon">
            <div data-testid="detail-title">{retroBoard.title}</div>
            {role !== 'developer' && <button type="button" onClick={editRetroHandler} data-testid="edit-icon"><EditIcon sx={{ fontSize: 30, fontWeight: 900 }} /></button>}
            {/* <button type="button" onClick={editRetroHandler}><EditIcon sx={{ fontSize: 30, fontWeight: 900 }} /></button> */}

          </div>
          <button type="button" className="cancel-button-retro" onClick={handleClick} data-testid="close-detail"><CloseRoundedIcon sx={{ fontSize: 40, fontWeight: 900 }} /></button>
        </div>
        <div className="retro-form-body-details">
          <div className="image">
            <img
              src={imageSrc}
              alt="retro"
              onError={handleImageError}
              data-testid="detail-image"
            />
          </div>
          <vr />
          <div className="retro-details-right">
            <h1>ACTION ITEMS</h1>

            <ul className="action-items-details" data-testid="detail-action-items">
              {retroBoard.action_items.map((actions) => (
                <li className="actions">{actions}</li>
              ))}
            </ul>
          </div>
        </div>
      </div>
      {edit && <RetroUpdate id={retroBoard.id} retroBoard={retroBoard} edit={edit} setEdit={setEdit} click={click} setClick={setClick} />}
    </div>
  );
}
RetroDetails.propTypes = {
  retroBoard: PropTypes.shape({
    title: PropTypes.string.isRequired,
    image_path: PropTypes.string.isRequired,
    action_items: PropTypes.arrayOf(PropTypes.string).isRequired,
    id: PropTypes.number.isRequired
  }).isRequired,
  click: PropTypes.func.isRequired,
  setClick: PropTypes.func.isRequired,
  role: PropTypes.string.isRequired
};

export default RetroDetails;
