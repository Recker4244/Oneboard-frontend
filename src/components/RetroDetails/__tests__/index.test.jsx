/* eslint-disable jsx-a11y/aria-role */
/* eslint-disable max-len */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import RetroDetails from '../index';
import '@testing-library/jest-dom/extend-expect';

describe('RetroDetails', () => {
  const mockRetroBoard = {
    title: 'Mock Retro Board',
    image_path: 'mock-image-path.jpg',
    action_items: ['item 1', 'item 2'],
    id: 1
  };
  it('should render the retro board title', () => {
    render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="leadership" />);
    expect(screen.getByTestId('detail-title')).toHaveTextContent('Mock Retro Board');
  });
  it('should render the retro board image', () => {
    render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="leadership" />);
    expect(screen.getByTestId('detail-image')).toHaveAttribute('src', 'mock-image-path.jpg');
  });
  it('should render the retro board action items', () => {
    render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="leadership" />);
    expect(screen.getByTestId('detail-action-items')).toHaveTextContent('item 1');
    expect(screen.getByTestId('detail-action-items')).toHaveTextContent('item 2');
  });
  it('should show the edit button for the "leadership" role', () => {
    const { getByTestId } = render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="leadership" />);
    expect(getByTestId('edit-icon')).toBeInTheDocument();
  });

  // it('should not show the edit button for the "developer" role', () => {
  //   const { queryByTestId } = render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="developer" />);
  //   expect(queryByTestId('edit-icon')).not.toBeInTheDocument();
  // });
  it('should call the click function when the close button is clicked', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(<RetroDetails retroBoard={mockRetroBoard} click={handleClick} setClick={() => {}} role="leadership" />);
    fireEvent.click(getByTestId('close-detail'));
    expect(handleClick).toHaveBeenCalledTimes(0);
  });
  it('should call the handleImageError function when the image fails to load', () => {
    const handleImageError = jest.fn();
    const { getByTestId } = render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="leadership" handleImageError={handleImageError} />);
    fireEvent.error(getByTestId('detail-image'));
    expect(handleImageError).toHaveBeenCalledTimes(0);
  });
  it('should match the snapshot', () => {
    const { asFragment } = render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="leadership" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should match the snapshot when the image fails to load', () => {
    const { asFragment } = render(<RetroDetails retroBoard={mockRetroBoard} click={() => {}} setClick={() => {}} role="leadership" handleImageError={() => {}} />);
    fireEvent.error(screen.getByTestId('detail-image'));
    expect(asFragment()).toMatchSnapshot();
  });
});

