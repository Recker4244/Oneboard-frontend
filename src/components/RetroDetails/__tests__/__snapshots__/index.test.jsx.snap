// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetroDetails should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="retro-form z-50"
  >
    <div
      class="retro-form-content"
    >
      <div
        class="retro-form-heading"
      >
        <div
          class="retro-title-left-icon"
        >
          <div
            data-testid="detail-title"
          >
            Mock Retro Board
          </div>
          <button
            data-testid="edit-icon"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gm5n8l-MuiSvgIcon-root"
              data-testid="EditIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </button>
        </div>
        <button
          class="cancel-button-retro"
          data-testid="close-detail"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1vj2ruk-MuiSvgIcon-root"
            data-testid="CloseRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.3 5.71a.9959.9959 0 0 0-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
            />
          </svg>
        </button>
      </div>
      <div
        class="retro-form-body-details"
      >
        <div
          class="image"
        >
          <img
            alt="retro"
            data-testid="detail-image"
            src="mock-image-path.jpg"
          />
        </div>
        <vr />
        <div
          class="retro-details-right"
        >
          <h1>
            ACTION ITEMS
          </h1>
          <ul
            class="action-items-details"
            data-testid="detail-action-items"
          >
            <li
              class="actions"
            >
              item 1
            </li>
            <li
              class="actions"
            >
              item 2
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RetroDetails should match the snapshot when the image fails to load 1`] = `
<DocumentFragment>
  <div
    class="retro-form z-50"
  >
    <div
      class="retro-form-content"
    >
      <div
        class="retro-form-heading"
      >
        <div
          class="retro-title-left-icon"
        >
          <div
            data-testid="detail-title"
          >
            Mock Retro Board
          </div>
          <button
            data-testid="edit-icon"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gm5n8l-MuiSvgIcon-root"
              data-testid="EditIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </button>
        </div>
        <button
          class="cancel-button-retro"
          data-testid="close-detail"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1vj2ruk-MuiSvgIcon-root"
            data-testid="CloseRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.3 5.71a.9959.9959 0 0 0-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
            />
          </svg>
        </button>
      </div>
      <div
        class="retro-form-body-details"
      >
        <div
          class="image"
        >
          <img
            alt="retro"
            data-testid="detail-image"
            src="https://st3.depositphotos.com/23594922/31822/v/600/depositphotos_318221368-stock-illustration-missing-picture-page-for-website.jpg"
          />
        </div>
        <vr />
        <div
          class="retro-details-right"
        >
          <h1>
            ACTION ITEMS
          </h1>
          <ul
            class="action-items-details"
            data-testid="detail-action-items"
          >
            <li
              class="actions"
            >
              item 1
            </li>
            <li
              class="actions"
            >
              item 2
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
